### PR TITLE
ARROW-10755: [Rust] [Parquet] Add support for writing boolean type

### PR DESCRIPTION
**Pull request by ghost**
Built this based on what I saw existing for the numeric implementations. Most of it was already there.

### CODE REVIEW

File: rust/parquet/src/arrow/arrow_writer.rs [hunk 1]
```diff
@@ -665,6 +670,16 @@ where
     values
 }
 
+fn get_bool_array_slice(array: &arrow_array::BooleanArray) -> Vec<bool> {
+    let mut values = Vec::with_capacity(array.len() - array.null_count());
```
Comment (nevi-me) [inline rust/parquet/src/arrow/arrow_writer.rs:674]: The logic that we were using was incorrect (at least once we get to lists). #8792 fixes this, so if it gets merged first, we would need to change this to populate values correctly .

**Review by nevi-me — APPROVED** [review]
Thanks @wjones1 

@alamb this is a low-risk change, and I need to update the nested struct level calculation if it gets merged first, so I'm happy that we merge it soon as someone else's reviewed it, and CI is happy.

File: rust/parquet/src/arrow/arrow_writer.rs [hunk 2]
```diff
@@ -1023,9 +1038,6 @@ mod tests {
     }
 
     #[test]
-    #[should_panic(
```
Comment (alamb) [inline rust/parquet/src/arrow/arrow_writer.rs:1026]: 👍

**Review by alamb — APPROVED** [review]
LGTM

**Pull request closed without merging**